cpp/src/io/parquet/writer_impl.cu

```diff
@@ -656,11 +686,12 @@ std::unique_ptr<std::vector<uint8_t>> writer::impl::write(
   const table_metadata *metadata,
   bool return_filemetadata,
   const std::string &column_chunks_file_path,
-  bool int96_timestamps,
+  std::vector<uint8_t> *decimal_precisions,
   rmm::cuda_stream_view stream)
 {
   pq_chunked_state state{metadata, SingleWriteMode::YES, int96_timestamps, stream};
 
+  if (decimal_precisions != nullptr) { state.decimal_precisions = *decimal_precisions; }
```

Comment:
is the condition needed here?

Reply:
Yes, without it we could dereference a null pointer.

Reply:
Ah the state actually has a copy, it's not always a pointer. I missed this. Disregard :)